File: AspNetMvcUnleashed/0672329980_source/Chapter07/CS/MvcApplication1/MvcApplication1/Controllers/ProductController.cs
using System.Web.Mvc;
using MvcApplication1.Models;

namespace MvcApplication1.Controllers
{
    public class ProductController : Controller
    {

        //
        // GET: /Product/Create

        public ActionResult Create()
        {
            return View();
        } 

        //
        // POST: /Product/Create

        [AcceptVerbs(HttpVerbs.Post)]
        public ActionResult Create(Product productToCreate)
        {
            // Add product to database
            return RedirectToAction("Index");
        }
    }
}

File: AspNetMvcUnleashed/0672329980_source/Chapter07/VB/MvcApplication1/MvcApplication1/Controllers/ProductController.vb
Public Class ProductController
    Inherits System.Web.Mvc.Controller

    Function Index() As ActionResult
        Return View()
    End Function


    ' GET: /Product/Create
    Function Create() As ActionResult
        Return View()
    End Function

    ' POST: /Product/Create
    <AcceptVerbs(HttpVerbs.Post)> _
    Function Create(ByVal productToCreate As Product) As ActionResult

        ' Add product to database
        Return RedirectToAction("Index")
    End Function

End Class
